### PR TITLE
Fix issue with getElementById

### PR DESCRIPTION
**Pull request by arv**
`querySelector('#id')` only allows valid CSS IDs but `getElementById` should allow any character but "HTML space" characters.

### CODE REVIEW

**Comment by arv** [comment]
@jmesserly

File: src/wrappers/ShadowRoot.js [hunk 1]
```diff
@@ -56,7 +58,9 @@
     },
 
     getElementById: function(id) {
-      return this.querySelector('#' + id);
+      if (spaceCharRe.test(id))
+        return null;
+      return this.querySelector('[id="' + id + '"]');
```
Comment (jmesserly) [inline src/wrappers/ShadowRoot.js:63]: I think there's a corner case here with double quotes in the ID. Probably no one will ever hit it though since quotes in IDs is crazy :)
Whether it's worth the cost of escaping to fix, I'm not sure. Probably not.

``` js
# at a JS console
var d = document.createElement('div')
d.innerHTML = '<div id=\'4"2\'>'
document.body.appendChild(d)
document.querySelector('[id="4"2"]')
# SyntaxError: Failed to execute query: '[id="4"2"]' is not a valid selector.
document.querySelector('[id="4\\"2"]')
# returns: <div id=​"4"2">​</div>​
```

**Comment by jmesserly** [comment]
LGTM!

File: src/wrappers/ShadowRoot.js (hunk 1 above)
Comment (arv) [inline src/wrappers/ShadowRoot.js:63]: I thought of this too but decide not to support it. We can always add a `replace(/"/g, '\\\\"')` later.
